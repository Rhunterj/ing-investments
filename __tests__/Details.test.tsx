
import { render, screen } from '@testing-library/react'
import { act } from 'react-dom/test-utils';
import StockDetailPage from '../pages/details/[slug]'
import { StockItemMock } from './__mocks__/StockItemMock';

describe('DetailPage', () => {
  it('renders a heading', () => {
    render(<StockDetailPage pageData={StockItemMock}/>);

    const heading = screen.getByRole('heading', {
      name: 'Adyen',
    })

    expect(heading).toBeInTheDocument()
  });

  
  it('renders without data', () => {
    act(() => {
      render(<StockDetailPage pageData={undefined}/>)
    })

    const heading = screen.getByText('There is no data available for this stock')

    expect(heading).toBeInTheDocument()
  });

  it('render Details unchanged', () => {
    const { container } = render(<StockDetailPage pageData={StockItemMock} />);
    expect(container).toMatchSnapshot();
  });
})