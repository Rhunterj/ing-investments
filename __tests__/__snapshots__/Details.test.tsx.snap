// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage render Details unchanged 1`] = `
<div>
  <div
    class="sc-6552922b-0 dVnnEP"
  >
    <main
      class="sc-514c9be-0 goCZuh"
    >
      <h1>
        Adyen
      </h1>
      <a
        href="/"
      >
        Back to overview
      </a>
      <table
        class="sc-ff8b6f11-0 hISlOf"
      >
        <tbody
          class="sc-ff8b6f11-1 iPdmmt"
        >
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Symbol
              </b>
               
              ADY
            </td>
          </tr>
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                CurrentValue
              </b>
               
              1622.365
            </td>
          </tr>
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Net change
              </b>
               
              +26.165
              /
              +1.639
            </td>
          </tr>
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Isin
              </b>
               
              NL0012969182
            </td>
          </tr>
        </tbody>
        <tbody
          class="sc-ff8b6f11-1 iPdmmt"
        >
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Previous close
              </b>
               
              1596.2
            </td>
          </tr>
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Highest price
              </b>
               
              1660.84
            </td>
          </tr>
          <tr
            class="sc-ff8b6f11-2 VICmL"
          >
            <td
              class="sc-ff8b6f11-3 eupGeZ"
            >
              <b
                class="sc-ff8b6f11-4 lbZLaJ"
              >
                Lowest price
              </b>
               
              1593.3
            </td>
          </tr>
        </tbody>
      </table>
    </main>
  </div>
</div>
`;
